### DISCUSSION TITLE
Using next.js API to proxify FastAPI

### DISCUSSION THREAD

**Discussion by matt-grain** [reflex-dev/reflex#120] [Q&A]
Hi,

If I'm not wrong currently the frontend is calling the fastapi backend directly so 2 ports need to be open on internet. And right now the FastAPI docs are available to everyone.

Would it be possible to use next.js API routing to hide the fastapi backend (which could be deployed on the same machine or elsewhere without CORS issues) and that all frontend calls will be managed by next.js ?

Due to this 2 ports constraints, I'm struggling to deploy my app to my current cloud provider (without using a VM)

Thanks !

**Comment by picklelo** [comment]
I hadn't considered this before, but it should be possible!

So if I understand correctly, you would want to:
* run the FastAPI on a port that is not exposed to the Internet (either on the same machine or another)
* run the Next app (web + api) on another port which *is* exposed to the Internet
* In the NextJS API routes, just proxy the requests to the FastAPI server and send it back the client
which would give us the advantage of only having to expose one port publicly.

I just created an [issue](https://github.com/pynecone-io/pynecone/issues/121) to investigate this.

**Reply by matt-grain** [reply]
yes :) I cannot summarize better :)

Practically, currently I'm using Azure App Service to deploy my pynecone.io app to check/learn how it can be deployed. I have defined a Docker image which is deployed automatically from Github to App Service. But for some reasons, App Service can only expose one port. So 3000 is mapped to 80 (or 443 whatever) but then the frontend try to call FastAPI on port 8000 and Azure doesn`t let it happen.

But anyway, I think hiding the FastAPI is not a bad idea (thanks for the ping end point ;-))

**Reply by picklelo** [reply]
Got it, makes sense - we will add this :) 
We can discuss further in the issue thread
